lib/Sema/TypeChecker.cpp

```diff
@@ -626,6 +626,8 @@ void swift::performTypeChecking(SourceFile &SF, TopLevelContext &TLC,
                                 unsigned ExpressionTimeoutThreshold) {
   if (SF.ASTStage == SourceFile::TypeChecked)
     return;
+  
+  SharedTimer timer("Type checking");
```

Comment:
There's not much point in adding together the various timers below; they're very separate tasks.

Reply:
…although arguably the name-binding one belongs inside `performNameBinding`, so it counts time spent on other files as well.